src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -162,6 +149,18 @@ public static Container combine(@CheckForNull Container parent, @Nonnull Contain
         return combined;
     }
 
+    private static Quantity safeGet(Container parent, Container template,
+                                    Function<ResourceRequirements, Map<String, Quantity>> resourceTypeMapper,
+                                    String field) {
+        Quantity templateField = resourceTypeMapper.apply(template.getResources()).get(field);
+        if (templateField != null && !Strings.isNullOrEmpty(templateField.getAmount())) {
```

Comment:
would be clearer with `Optional.ofNullable` as in https://github.com/jenkinsci/kubernetes-plugin/pull/311/files#diff-eff844000c9e60765310e70fa8c8a71cR208 ?

Reply:
@carlossg thanks for the feedback.  I've addressed this in the latest commit.